packages/form/addon/components/cf-field/input/file.js

```diff
@@ -4,64 +4,87 @@ import Component from "@glimmer/component";
-import getFileAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/fileanswer-info.graphql";
+import getFilesAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/filesanswer-info.graphql";
 
 export default class CfFieldInputFileComponent extends Component {
   @service intl;
 
   @queryManager apollo;
 
-  get downloadUrl() {
-    return this.args.field?.answer?.value?.downloadUrl;
-  }
-
-  get downloadName() {
-    return this.args.field?.answer?.value?.name;
+  get files() {
+    return this.args.field?.answer?.value;
   }
 
   @action
-  async download() {
-    const { downloadUrl } = await this.apollo.query(
+  async download(fileName) {
+    if (!fileName) {
+      return;
+    }
+    const answers = await this.apollo.query(
       {
-        query: getFileAnswerInfoQuery,
+        query: getFilesAnswerInfoQuery,
         variables: { id: this.args.field.answer.raw.id },
         fetchPolicy: "network-only",
       },
-      "node.fileValue"
+      "node.value"
     );
 
+    const { downloadUrl } = answers.find((file) => file.name === fileName);
     if (downloadUrl) {
       window.open(downloadUrl, "_blank");
     }
   }
 
   @action
   async save({ target }) {
-    const file = target.files[0];
+    // store the old list of files
+    const fileList = [...(this.files || [])];
 
-    if (!file) {
+    // unwrap files from FileList construct
+    const newFiles = [];
+    for (let i = 0; i < target.files.length; i++) {
+      newFiles.push({ name: target.files[i].name });
+      if (
+        !fileList.find((existing) => existing.name === target.files[i].name)
+      ) {
+        fileList.push({ name: target.files[i].name });
+      }
+    }
+    if (!newFiles) {
       return;
     }
 
-    const { fileValue } = await this.args.onSave(file.name);
+    // trigger save action for deduped file list of old and new files reduces properties
+    const { filesValue } = await this.args.onSave(
+      fileList.map(({ name, id }) => ({ name, id }))
+    );
 
     try {
-      const response = await fetch(fileValue.uploadUrl, {
-        method: "PUT",
-        body: file,
-      });
+      // iterate over list of new files and map to graphql answer values
+      // this way we only upload new files selected by the input
+      const _newFiles = newFiles.map((file) =>
+        filesValue.find((value) => file.name === value.name)
+      );
+      for await (const file of _newFiles) {
+        const response = await fetch(file.uploadUrl, {
+          method: "PUT",
+          body: file,
+        });
 
-      if (!response.ok) {
-        throw new Error();
-      }
+        if (!response.ok) {
+          throw new Error();
+        }
 
-      this.args.field.answer.value = {
-        name: file.name,
-        downloadUrl: fileValue.downloadUrl,
-      };
+        // replace fileList value with filesValue value
+        fileList.splice(
+          fileList.findIndex((f) => f.name === file.name),
+          1,
+          file
+        );
```

Comment:
What exactly is this doing?

Reply:
This was my unnecessary approach to combine the graph-QL answer with the file list from the input. But actually we should set the input value to the actual graph-QL answer value. `Code deleted`